### ISSUE TITLE
mkl compatibility issue on windows

### ISSUE THREAD

**Issue by domfournier** [simpeg/pydiso#2]
The latest mkl release (2022.0) on conda-forge is making `pardiso` to fail on Windows with error:
`pydiso.mkl_solver.PardisoError: Analysis step error, reordering problem`

Fixing the version to 2021.4 does the trick ... for now.

https://anaconda.org/conda-forge/mkl/files

Thought I would flag it here.

**Comment by jcapriot** [comment]
Dom, it seems odd to me that it is actually throwing this error and not some other binary incompatibility error. I don't see anything that changed in how it gets linked though. Do other simpler functions work from this? i.e. `get_mkl_version()`?

Can you make a minimal example that works in 2021.4 that doesn't work in 2022.0 (Just a simple matrix that works in one version but not the other)?

**Comment by jcapriot** [comment]
I'm curious actually how you got to the point of having 2022 and pydiso from conda-forge. My recent tests actually block pydiso and mkl2022 from being installed together. I did just recently accept an automated PR on the conda-forge/pydiso-feedstock repo that controls the distribution there.

I've also tested this myself on my windows PC building pydiso from source against mkl 2022 and all my tests seem to pass.
